stl/inc/filesystem

```diff
@@ -3585,23 +3585,33 @@ namespace filesystem {
     }
 
     // FUNCTION remove
-    inline bool remove(const path& _Target) {
+    inline bool remove(const path& _Target, error_code& _Ec) noexcept {
         // remove file _Target (even if it is a directory); returns whether the file was removed
         // note !exists(_Target) is not an error, and merely returns false
-        const auto _Result = __std_fs_remove(_Target.c_str());
-        if (_Result._Error != __std_win_error::_Success) {
-            _Throw_fs_error("remove", _Result._Error, _Target);
+        auto _Result = __std_fs_remove(_Target.c_str());
+        if (_Result._Error == __std_win_error::_Access_denied) {
+            // Try to remove the read-only attribute and try again
+            const auto _Perm_change_err = __std_fs_change_permissions(_Target.c_str(), false, false);
+            if (_Perm_change_err != __std_win_error::_Success) {
+                _Ec = _Make_ec(_Perm_change_err);
+                return false;
+            }
+            _Result = __std_fs_remove(_Target.c_str());
         }
-
+        _Ec = _Make_ec(_Result._Error);
         return _Result._Removed;
     }
 
-    inline bool remove(const path& _Target, error_code& _Ec) noexcept {
+    inline bool remove(const path& _Target) {
         // remove file _Target (even if it is a directory); returns whether the file was removed
         // note !exists(_Target) is not an error, and merely returns false
-        const auto _Result = __std_fs_remove(_Target.c_str());
-        _Ec                = _Make_ec(_Result._Error);
-        return _Result._Removed;
+        error_code _Ec;
+        const auto _Result = _STD filesystem::remove(_Target, _Ec);
+        if (static_cast<__std_win_error>(_Ec.value()) != __std_win_error::_Success) {
+            _Throw_fs_error("remove", static_cast<__std_win_error>(_Ec.value()), _Target);

```

Comment:
While this is correct, I believe that it can be improved.

Calling `_Throw_fs_error` with `__std_win_error` will call `_Make_ec` again. Calling `_Throw_fs_error` with the `error_code` is less verbose and more efficient:

https://github.com/microsoft/STL/blob/ac4fde764ae716512bdd67e0e9577bd53947bc3d/stl/inc/filesystem#L1843-L1845
https://github.com/microsoft/STL/blob/ac4fde764ae716512bdd67e0e9577bd53947bc3d/stl/inc/filesystem#L1852-L1854

Additionally, the `error_code` can be directly tested to detect errors, as `is_empty` does above:

https://github.com/microsoft/STL/blob/ac4fde764ae716512bdd67e0e9577bd53947bc3d/stl/inc/filesystem#L3527-L3528

So this can follow the same pattern:

```suggestion
        if (_Ec) {
            _Throw_fs_error("remove", _Ec, _Target);
```